velox/exec/tests/PlanBuilderTest.cpp

```diff
@@ -250,4 +254,60 @@ TEST_F(PlanBuilderTest, missingOutputType) {
       "outputType must be specified");
 }
 
+TEST_F(PlanBuilderTest, projectExpressions) {
+  // Non-typed Expression
+  auto data = ROW({"c0"}, {BIGINT()});
+  VELOX_CHECK_EQ(
+      PlanBuilder()
+          .tableScan("tmp", data)
+          .projectExpressions(
+              {std::make_shared<core::FieldAccessExpr>("c0", std::nullopt)})
+          .planNode()
+          ->toString(true, true, nullptr),
+      PlanBuilder()
+          .tableScan("tmp", data)
+          .project({"c0"})
+          .planNode()
+          ->toString(true, true, nullptr)); // simple field access
+  data = ROW({"c0"}, {ROW({"field0"}, {BIGINT()})});
+  VELOX_CHECK_EQ(
+      PlanBuilder()
+          .tableScan("tmp", data)
+          .projectExpressions({std::make_shared<core::FieldAccessExpr>(
+              "field0",
+              std::nullopt,
+              std::vector<core::ExprPtr>{
+                  std::make_shared<core::FieldAccessExpr>(
+                      "c0", std::nullopt)})})
+          .planNode()
+          ->toString(true, true, nullptr),
+      PlanBuilder()
+          .tableScan("tmp", data)
+          .project({"c0.field0"})
+          .planNode()
+          ->toString(
+              true,
+              true,
+              nullptr)); // dereference test using field access query
+
+  // Test Typed Expressions.
+  // PrestoQueryRunnerTest handles testing accuracy of the generated SQL using
+  // QueryRunner, here we want to simply verify that the overloaded function
+  // projectExpressions can generate a query plan using typed expressions.
+  // Untyped and typed expressions have slightly different plan node string
+  // outputs, making it difficult to test.
+  PlanBuilder()
+      .tableScan("tmp", ROW({"c0"}, {ROW({BIGINT()})}))
+      .projectExpressions(
+          {std::make_shared<core::FieldAccessTypedExpr>(BIGINT(), "c0")})
+      .planNode();
+  PlanBuilder()
+      .tableScan("tmp", ROW({"c0"}, {ROW({BIGINT()})}))
+      .projectExpressions({std::make_shared<core::FieldAccessTypedExpr>(
+          VARCHAR(),
+          std::make_shared<core::FieldAccessTypedExpr>(VARCHAR(), "c0"),
+          "field0")})
+      .planNode();
```

Comment:
Let's also check that these plans are as expected too, e.g., by checking the toString() result equals to an expected string.